### PR TITLE
Add a GitHub action to build on Linux and macOS

### PR DESCRIPTION
**Pull request by skitt**
As with Travis CI, the test is only deemed successful on each platform
if the Stella build completes successfully.

For Linux, the test runs on Ubuntu with the toolchain test build
repository, using G++ 9, and the packaged version of SDL2, which is
guaranteed to be at least 2.0.10.

For macOS, SDL2 2.0.10 is still built from the upstream source code;
however that no longer builds with the newer Xcode versions available
in GHAs, so the build now uses "Unix-style" ./configure && make.

The runners provide two cores on Linux and three cores on macOS, so
the compile stages use the appropriate -j setting to build in
parallel.

Fixes: #742 
Signed-off-by: Stephen Kitt <steve@sk2.org>

### CODE REVIEW

**Comment by skitt** [comment]
GHAs need to be enabled in the project settings.

**Comment by skitt** [comment]
See https://github.com/skitt/stella/actions/runs/3217767267/jobs/5261124666 for a successful run on Linux.

**Comment by skitt** [comment]
See https://github.com/skitt/stella/actions/runs/3217975817 for a successful non-parallel build on both platforms, and https://github.com/skitt/stella/actions/runs/3218052527 for a successful parallel build on both platforms.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        run: |
+          sudo add-apt-repository ppa:ubuntu-toolchain-r/test
+          sudo apt update
+          sudo apt install g++-9
+      - name: Install SDL2
+        run: |
+          curl https://www.libsdl.org/release/SDL2-2.0.10.tar.gz | tar xzf -
+          mkdir SDL2-2.0.10/build
+          cd SDL2-2.0.10/build
+          ../configure && make -j2 && sudo make install
```
Comment (skitt) [inline .github/workflows/build.yml:26]: On Ubuntu, we could save a lot of time by installing `libsdl2-dev` instead (the default release of Ubuntu on GHAs, 20.04 currently, has SDL 2.0.10); but that would tie the build to whatever is the default on whatever release of Ubuntu we get at a given point in time.

Comment (sa666666) [reply .github/workflows/build.yml:26]: That would be fine.  As I generally use Ubuntu for development anyway, there is no issue with tying it.  Please add this if you like.

Comment (skitt) [reply .github/workflows/build.yml]: Done!

**Comment by sa666666** [comment]
Thanks for implementing this!

**Pull request merged**